web/client/api/CSW.js

```diff
@@ -197,41 +198,67 @@ const addCapabilitiesToRecords = (_dcRef, result, options) => {
         return result;
     }
     const { value: _url } = _dcRef?.find(t =>
-        REGEX_WMS_ALL.some(regex=> t?.scheme?.match(regex) || t?.protocol?.match(regex))) || {}; // Get WMS URL from references
+        REGEX_WMS_ALL.some(regex => t?.scheme?.match(regex) || t?.protocol?.match(regex))) || {}; // Get WMS URL from references
     const [parsedUrl] = _url && _url.split('?') || [];
-    if (!parsedUrl) return {...result}; // Return record when no url found
+    if (!parsedUrl) return { ...result }; // Return record when no url found
 
     const cached = capabilitiesCache[parsedUrl];
     const isCached = !isEmpty(cached);
     return Promise.resolve(
         isCached
             ? cached
             : WMS.getCapabilities(parsedUrl + '?version=')
-                .then((caps)=> WMS.flatLayers(caps.Capability))
-                .catch(()=> []))
+                .then((caps) => WMS.flatLayers(caps.Capability))
+                .catch(() => []))
         .then((layers) => {
             if (!isCached) {
                 capabilitiesCache[parsedUrl] = layers;
             }
             // Add visibility limits scale data of the layer to the record
             return {
                 ...result,
-                records: result?.records?.map(record=> {
+                records: result?.records?.map(record => {
                     const name = get(getLayerReferenceFromDc(record?.dc, null, false), 'params.name', '');
                     const {
                         MinScaleDenominator,
                         MaxScaleDenominator
-                    } = layers.find(l=> l.Name === name) || {};
+                    } = layers.find(l => l.Name === name) || {};
                     return {
                         ...record,
                         ...((!isNil(MinScaleDenominator) || !isNil(MaxScaleDenominator))
-                        && {capabilities: {MaxScaleDenominator, MinScaleDenominator}})
+                            && { capabilities: { MaxScaleDenominator, MinScaleDenominator } })
                     };
                 })
             };
         });
 };
-
+/**
+ * handle getting bbox from capabilities in case of 3D tile layer for CSW records
+ * @param {object} result csw results object
+ * @return {object} csw records
+ */
+const getBboxFor3DLayersToRecords = async(result)=> {
+    if (!result) return result;
+    let { records } = result;
+    if (records?.length) {
+        let records3D = records.filter(rec=>rec?.dc?.format === THREE_D_TILES);
```

Comment:
hi, there is a mistake here that could have been fetched from adding proper unit testing

<hr>
you cannot filter records here and then update them layer on because you can end updating the wrong one:

examples that you must have covered during tests
- you have 4 records, the first one is a THREE_D_TILES
- you have 4 records, the last one is a THREE_D_TILES
- you have 3 records, the middle one is a THREE_D_TILES

in second and third scenario you are going to update the extent to the wrong layer

avoid filtering and return the getCapabilities only if you match the config otherwise you can return a a promise that resolves the original record for those that are not THREE_D_TILES